kernel/src/callback.rs

```diff
@@ -6,41 +6,80 @@ pub struct AppId {
     idx: usize,
 }
 
+/// The kernel can masquerade as an app. IDs >= this value are the kernel.
+/// These IDs are used to identify which kernel container is being accessed.
+const KERNEL_APPID_BOUNDARY: usize = 100;
+
 impl AppId {
     pub fn new(idx: usize) -> AppId {
         AppId { idx: idx }
     }
 
+    pub const fn kernel_new(idx: usize) -> AppId {
+        AppId { idx: idx }
```

Comment:
Ahhh... right.. I remember now. Can't have those checks and be a `const fn`, which we really want to facilitate static initialization.

Eventually refinement types ( https://github.com/rust-lang/rfcs/issues/671 ) could fix this problem, but for now there's not a great way to handle this.